src/lib.rs

```diff
@@ -251,10 +263,15 @@ impl Builder {
               state.maximized = is_maximized;
               state.fullscreen = is_fullscreen;
 
+              let mut scale_factor = 1f64;
+              if let Some(monitor) = window_clone.current_monitor().unwrap() {
+                scale_factor = monitor.scale_factor();
+              }
+
               // It doesn't make sense to save a window with 0 height or width
               if size.width > 0 && size.height > 0 && !is_maximized {
-                state.width = size.width;
-                state.height = size.height;
+                state.width = size.width.scale(scale_factor);
+                state.height = size.height.scale(scale_factor);
```

Comment:
why you need to scale this size, it is already scaled?

Reply:
The `state` always holds the windows width/height after the scaling factor has been applied, to store the "unscaled" dimensions. The unscaled dimensions are then scaled using the (target) monitors scaling factor when `set_size` is called.

Without these two lines, the wrong dimensions will be saved when resizing the window.

Reply:
Hmmm, that shouldn't be the case since the Resized event returns physical units, not logical ones (so the reported size should always be the same no matter the scale factor)

Reply:
That's interesting. I removed these two lines to test my assumptions and the window restored to 1/2 the correct size when going from an external monitor (scale 1) to the laptop monitor (scale 2). I could be missing something else though!

Reply:
As Fabian said, it shouldn't really matter what scale factor, not when saving and not when storing since we are dealing with physical sizes.

Reply:
Could you try to record the size changes you mention with your phone? open a window on external monitor, resize it, take a picture of it and close it. Then disconnect the external monitor and let the window open on the laptop monitor and take a picture of it

Reply:
Looking this over with fresh eyes, you're correct that the scaling shouldn't be applied here.

Applying the scaling here was masking the fact that when changing monitors `m.name().map(ToString::to_string).unwrap_or_default() == state.monitor` evaluates to `false`, so the `scale_factor = m.scale_factor()` from within that block was never being called when changing monitors.

I moved this scaling out of this conditional block and reverted to using `self.current_monitor()`, but that goes against you comment above of using the target monitor not the current monitor?

This branch has been rebased on `dev` and functions as you would expect now. Repeated opens on the same monitor show no changes in size, and the window opens with the same size when switching between the external/laptop monitor.

Reply:
> Could you try to record the size changes you mention with your phone? open a window on external monitor, resize it, take a picture of it and close it. Then disconnect the external monitor and let the window open on the laptop monitor and take a picture of it

The plugin should be storing a physical size and restoring a physical size so there is no need to scale anything. Could you please give the above info?